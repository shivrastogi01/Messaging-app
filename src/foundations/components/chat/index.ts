export * from "./text-message/text-message";
export * from "./image-message/image-message";
export * from "./link-message/link-message";
export * from "./giphy-message/giphy-message";
export * from "./avatar/avatar";
