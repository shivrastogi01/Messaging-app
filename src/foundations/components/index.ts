export * from "./presentation";
export * from "./layout";
