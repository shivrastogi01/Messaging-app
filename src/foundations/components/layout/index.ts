export * from "./styled-box/styled-box";
export * from "./scroll-view/scroll-view";
export * from "./flex-row/flex-row";
export * from "./flex-column/flex-column";
export * from "./modal/modal";
export * from "./drawer/drawer";
export * from "./list-item/list-item";
export * from "./dropdown/dropdown";
