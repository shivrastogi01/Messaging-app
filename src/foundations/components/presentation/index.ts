export * from "./heading/heading";
export * from "./label/label";
export * from "./title/title";
export * from "./icon/icon";
export * from "./button/button";
export * from "./input/input";
export * from "./textarea/textarea";
