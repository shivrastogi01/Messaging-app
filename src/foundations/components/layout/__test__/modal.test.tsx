import React from "react";
import renderWithTheme from "main/testing/render-with-theme";
import { Modal } from "foundations/components/layout";

describe("Modal", () => {
  test("closed variant matches the snapshot", () => {
    renderWithTheme(<Modal>Modal content</Modal>);
    expect(document.body).toMatchSnapshot();
  });

  test("opened variant matches the snapshot", () => {
    renderWithTheme(<Modal open>Modal content</Modal>);
    expect(document.body).toMatchSnapshot();
  });
});
