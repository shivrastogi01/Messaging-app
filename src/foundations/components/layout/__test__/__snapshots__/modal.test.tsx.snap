// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal closed variant matches the snapshot 1`] = `
<body>
  <div />
</body>
`;

exports[`Modal opened variant matches the snapshot 1`] = `
<body>
  @media screen and (min-width:600px) {

}

@media screen and (min-width:1200px) {

}

@media screen and (min-width:600px) {

}

<div />
  .c0 {
  background: #FFFFFF;
  bottom: 0;
  height: 100%;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  width: 100%;
  z-index: 200;
}

.c1 {
  background: #FFFFFF;
  border-radius: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-height: 100%;
  padding: 0;
  width: 100%;
  z-index: 300;
}

@media screen and (min-width:600px) {
  .c0 {
    padding: 10vh 15%;
    background-color: rgba(0,0,0,0.5);
  }
}

@media screen and (min-width:1200px) {
  .c0 {
    padding: 10vh 25%;
  }
}

@media screen and (min-width:600px) {
  .c1 {
    border-radius: 10px;
    box-shadow: 0 4px 30px rgba(0,0,0,0.4);
    padding: 40px;
  }
}

<div
    class="c0"
  >
    <section
      class="c1"
      style="transform: none;"
    >
      Modal content
    </section>
  </div>
</body>
`;
